Tasks/CopyFiles/task.json

```diff
@@ -0,0 +1,87 @@
+﻿{
+  "id": "5BFB729A-A7C8-4A78-A7C3-8D717BB7C13C",
+  "name": "CopyFiles",
+  "friendlyName": "Copy Files",
+  "description": "Copy files from source folder to target folder using minimatch patterns",
+  "helpMarkDown": "[More Information](http://go.microsoft.com/fwlink/?LinkID=)",
+  "category": "Utility",
+  "visibility": [
+    "Build"
+  ],
+  "author": "Microsoft Corporation",
+  "version": {
+    "Major": 1,
+    "Minor": 0,
+    "Patch": 0
+  },
+  "demands": [
+  ],
+  "minimumAgentVersion": "1.91.0",
+  "groups": [
+        {
+            "name": "advanced",
+            "displayName": "Advanced",
+            "isExpanded": false
+        }
+    ],
+  "inputs": [
+    {
+      "name": "SourceFolder",
+      "type": "string",
+      "label": "Source Folder",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": "Source folder to apply copy patterns to"
+    },  
+    {
+      "name": "Contents",
+      "type": "multiLine",
+      "label": "Contents",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": "File paths to include as part of the copy. Supports multiple lines or minimatch patterns. [More Information](http://go.microsoft.com/fwlink/?LinkID=)"
+    },
+    {
+      "name": "TargetFolder",
+      "type": "string",
+      "label": "Target Folder",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": "Target folder files will copy to"
+    },
+    {
+      "name": "CleanTargetFolder",
+      "type": "boolean",
+      "label": "Clean Target Folder",
+      "defaultValue": "false",
+      "required": false,
+      "helpMarkDown": "Delete all existing files in target folder before copy.",
+      "groupName":"advanced"
+    },
+    {
+      "name": "OverWrite",
+      "type": "boolean",
+      "label": "Over Write",
+      "defaultValue": "false",
+      "required": false,
+      "helpMarkDown": "Relace existing file in target folder",
+      "groupName":"advanced"
+    },
+    {
+      "name": "UseCommonRoot",
+      "type": "boolean",
+      "label": "Use Common Root",
+      "defaultValue": "false",
```

Comment:
Short the path on target folder by remove the common root of files need copied.
Ex.
source:  c:\a\b\c\test1.txt
              c:\a\b\c\test2.txt
              c:\a\b\c\test3.txt
SourceFolder = c:\a
TargetFolder = d:\t
With using common root.
copy result:
            d:\t\test1.txt
            d:\t\test2.txt
            d:\t\test3.txt
Otherwise:
            d:\t\b\c\test1.txt
            d:\t\b\c\test2.txt
            d:\t\b\c\test3.txt